kie-api/src/main/java/org/kie/api/builder/KieBuilder.java

```diff
@@ -39,6 +41,13 @@
     KieBuilder buildAll();
 
     /**
+     * Builds all the KieBases contained in the KieModule for which this KieBuilder has been created
+     * @param classFilters Used to prevent compilation of Java source files.
+     *          A Java source file vetoed by any of the given filters will not be compiled.
+     */
+    KieBuilder buildAll(Collection<KieClassFilter> classFilters);
```

Comment:
@mfusco true, `Predicate<String>` should work for all the resources (assuming we document and use it like that).

@mbarkley  correct, currently I am not aware of that additional requirement to filter other resources. But few days back I had no idea we would ever need filtering for java classes :) It was just an idea I am not going to push it....